feature_layers/add-feature-clustering/src/main/java/com/esri/samples/add_feature_clustering/AddFeatureClusteringSample.java

```diff
@@ -101,29 +110,69 @@ public void start(Stage stage) {
       var portal = new Portal("https://www.arcgis.com/");
       portalItem = new PortalItem(portal, "aa44e79a4836413c89908e1afdace2ea");
 
-      mMap = new ArcGISMap(portalItem);
+      map = new ArcGISMap(portalItem);
+
+      VBox vBoxPopupInformation = new VBox();
+      popupContentLabel = new Label();
+      setupVBoxStyling(vBoxPopupInformation);
+      vBoxPopupInformation.getChildren().addAll(popupContentLabel);
+      vBoxPopupInformation.setVisible(false);
 
       // Get the Zurich buildings feature layer once the map has finished loading
-      mMap.addDoneLoadingListener(() -> {
-        if (mMap.getLoadStatus() == LoadStatus.LOADED) {
-          LayerList l = mMap.getOperationalLayers();
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          LayerList l = map.getOperationalLayers();
           if (l.isEmpty())
             return;
-          mLayer = (FeatureLayer) mMap.getOperationalLayers().get(0);
+          layer = (FeatureLayer) l.get(0);
+
+          mapView.setOnMouseClicked(mouseEvent -> {
+            if (layer == null) {
+              return;
+            }
+            // Identify the tapped observation.
+            Point2D point = new Point2D(mouseEvent.getX(), mouseEvent.getY());
+            ListenableFuture<IdentifyLayerResult> identifiedLayerResults = mapView.identifyLayerAsync(layer, point, 3.0, true);
+            identifiedLayerResults.addDoneListener(() -> {
+              try {
+                // clear the list of popup content
+                popupContent = "";
+                IdentifyLayerResult layer = identifiedLayerResults.get();
+
+                for (Popup popup : layer.getPopups()) {
+                  final Map<String, Object> attributes = popup.getGeoElement().getAttributes();
+                  for (final String name : attributes.keySet()) {
+                    popupContent += name + ": " + attributes.get(name).toString() + "\n";
+                  }
+                }
+                System.out.println("###: " + popupContent);
+                Platform.runLater(() ->{
+                  popupContentLabel.setText(popupContent);
+                  popupContentLabel.setTextFill(Color.WHITE);
+                  vBoxPopupInformation.setVisible(!popupContentLabel.getText().isEmpty());
+                });
+
+              } catch (ExecutionException | InterruptedException e) {
+                throw new RuntimeException(e);
```

Comment:
Since this is inside a handler it might just get swallowed up by the calling thread, so maybe a `System.err.println` of the `getMessage` would be better.